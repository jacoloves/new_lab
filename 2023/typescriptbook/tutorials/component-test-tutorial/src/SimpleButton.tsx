import { useState } from "react";

export const SimpleButton: () => JSX.Element = () => {
  const [state, setState] = useState(true);
  const handleClick = () =>{
    setState((prevState) => !prevState);
  };
  return <button onClick={handleClick}>{state ? "ON" : "OFF"}</button>;
};
