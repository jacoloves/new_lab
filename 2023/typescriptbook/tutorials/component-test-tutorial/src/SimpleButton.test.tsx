import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { SimpleButton } from "./SimpleButton";

test("ボタンをクリックするとON/OFFの表示が切り替わる", async () => {
  const user = userEvent.setup();
  render(<SimpleButton />);
  const simpleButton = screen.getByRole("button");
  expect(simpleButton).toHaveTextContent("OFF");
  await user.click(simpleButton);
  expect(simpleButton).toHaveTextContent("ON");
});

test("描画されてすぐはOFFと表示されている", () => {
  const view = render(<SimpleButton />);
  expect(view.container).toMatchSnapshot();
});
