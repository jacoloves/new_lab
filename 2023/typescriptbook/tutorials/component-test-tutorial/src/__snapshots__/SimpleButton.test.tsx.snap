// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`描画されてすぐはOFFと表示されている 1`] = `
<div>
  <button>
    OFF
  </button>
</div>
`;
